File: src/main/java/cube8540/oauth/authentication/credentials/oauth/client/domain/OAuth2ClientRepository.java
package cube8540.oauth.authentication.credentials.oauth.client.domain;

import cube8540.oauth.authentication.users.domain.UserEmail;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface OAuth2ClientRepository extends JpaRepository<OAuth2Client, OAuth2ClientId> {

    Long countByClientId(OAuth2ClientId clientId);

    @EntityGraph(attributePaths = {"redirectUris", "grantTypes", "scopes"})
    Optional<OAuth2Client> findByClientId(OAuth2ClientId clientId);

    Page<OAuth2Client> findByOwner(UserEmail owner, Pageable pageable);
}

File: src/main/kotlin/cube8540/oauth/authentication/oauth/client/domain/OAuth2ClientRepository.kt
package cube8540.oauth.authentication.oauth.client.domain

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface OAuth2ClientRepository: JpaRepository<OAuth2Client, OAuth2ClientId> {

    fun countByClientId(clientId: OAuth2ClientId): Long

    @EntityGraph(attributePaths = ["redirectUris", "grantTypes", "scopes"])
    fun findByClientId(clientId: OAuth2ClientId): Optional<OAuth2Client>

    fun findByOwner(owner: ClientOwner, pageable: Pageable): Page<OAuth2Client>

}